### PR TITLE
RescheduleAppointments First commit

### PR DESCRIPTION
**Pull request by ibenitamar**

### CODE REVIEW

**Comment by ibenitamar** [comment]
re-opening after cla sign

**Comment by ibenitamar** [comment]
@maliroteh-sf created the rebase again

**Review by maliroteh-sf — COMMENTED** [review]
Finally finished the first take on this PR and left some comments for you. There are many instances of commented code left over across many files. Please remove all such instances to make the code cleaner before @khawkins takes a look at the PR, to make it easier on the reviewers.

Also you have several code files (.js, .cls) where the files are huge (several hundred or even thousand lines of code). It is difficult to code review such large files and even more difficult for external customers to digest RescheduleAppointments sample. Consider refactoring the code and breaking things up into utility/helper classes.

**Comment by ibenitamar** [comment]
> Also you have several code files (.js, .cls) where the files are huge (several hundred or even thousand lines of code). It is difficult to code review such large files and even more difficult for external customers to digest RescheduleAppointments sample. Consider refactoring the code and breaking things up into utility/helper classes.

@maliroteh-sf Thanks for you comments! regarding the length of the files and in general this code was not written by me but taken from the appointment assistant code (as requested) and was just adjusted to mobile + added assign to mobile worker capabilities (to be able to provide this highly requested capability to the mobile tech). I agree with you on those points I created a doc with needed refactors as technical debt and my team will eventually address  them as we go along. 
re. missing tests this is a user story for 244 to complete.

File: RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingUtils/test.js
Comment (ibenitamar) [reply RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingUtils/test.js:1]: @maliroteh-sf forgot to delete this

File: RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingUtils/rebookingDateCalculationsHelpers
Comment (ibenitamar) [reply RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingUtils/rebookingDateCalculationsHelpers:2]: @maliroteh-sf Forgot to delete this I was planning to refactor all the calc helpers from the components but it's in progress and not relevant for 242

File: RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingSlotsContainer/__tests__/mobileAppointmentBookingSlotsContainer.test.js
Comment (ibenitamar) [reply RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingSlotsContainer/__tests__/mobileAppointmentBookingSlotsContainer.test.js]: Removed folder

**Comment by maliroteh-sf** [comment]
@ibenitamar  Ok sounds good... looking forward to seeing the code refactoring and test coverage PRs coming in later.

FYI I see that you've addressed some of my feedback on removing commented code but there still are many other locations with commented code. Please review all of my feedback and address all of the files by removing dead/commented code. I didn't mark a feedback on each instance of commented code since there are many instances so just review all of your code files please.

Other than that, I'm good with this PR but we'd need to wait for @khawkins to review it too before it can be merged.

File: RescheduleAppointments/.gitignore
Comment (khawkins) [inline RescheduleAppointments/.gitignore:11]: Please remove this line from `.gitignore` (it can stay in `.forceignore`), and add the following `jsconfig.json` file to `force-app/main/default/lwc/`:

```json
{
    "compilerOptions": {
        "experimentalDecorators": true
    }
}
```

This will get rid of TS-1219 errors in VSCode whenever we have annotations in our LWCs like `@api`. See [this PR comment](https://github.com/forcedotcom/LWC-Mobile-Samples/pull/10#discussion_r1069882021) for more discussion.

File: RescheduleAppointments/force-app/main/default/customLables/customLables.labels-meta.xml
Comment (khawkins) [inline RescheduleAppointments/force-app/main/default/customLables/customLables.labels-meta.xml:2]: Nit: this directory path / file name structure contains a misspelling: `customLables` vs. `customLabels`. We should change all references to the latter.

File: RescheduleAppointments/force-app/main/default/lwc/aaRebookingCalendar/aaRebookingCalendar.html
Comment (khawkins) [inline RescheduleAppointments/force-app/main/default/lwc/aaRebookingCalendar/aaRebookingCalendar.html:1]: What is the "aa" in aaRebookingCalendar?

File: RescheduleAppointments/force-app/main/default/lwc/aaRebookingCalendar/aaRebookingCalendar.html
Comment (khawkins) [inline RescheduleAppointments/force-app/main/default/lwc/aaRebookingCalendar/aaRebookingCalendar.html:22]: There are many components in this project that generate Mobile SLDS errors for missing labels:

> Clickable images, Lightning button icons, and form elements require labels for mobile users. SLDS(4)

We should add labels to these controls.

image

File: RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingCustomToast/mobileAppointmentBookingCustomToast.html
Comment (khawkins) [inline RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingCustomToast/mobileAppointmentBookingCustomToast.html:11]: This line generates the following SLDS error:

> Deprecated, or invalid, utility class: "slds-notify__content" SLDS(0)

File: RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingSlotsContainer/mobileAppointmentBookingSlotsContainer.js
Comment (khawkins) [inline RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingSlotsContainer/mobileAppointmentBookingSlotsContainer.js:123]: The naming conventions of e.g. this getter is inconsistent with others in the file (like `onWeekUpdated()` above).

**Review by khawkins — CHANGES_REQUESTED** [review]
In addition to the individual comments, please rebase against the HEAD of `main`.

I just skimmed the content of the updates. There's a lot of code there to cover, so I don't imagine I got a good deep dive against it.

File: RescheduleAppointments/force-app/main/default/lwc/aaRebookingCalendar/aaRebookingCalendar.html
Comment (ibenitamar) [reply RescheduleAppointments/force-app/main/default/lwc/aaRebookingCalendar/aaRebookingCalendar.html:1]: It's for appointment assistant where the code is taken from. Originally it was meant to consume the actual component but since those are not yet developed on core I guess I should rename the component for the samples project.

File: RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingCustomToast/mobileAppointmentBookingCustomToast.html
Comment (ibenitamar) [reply RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingCustomToast/mobileAppointmentBookingCustomToast.html:11]: @khawkins it's is still supported and in 244 the toast component will be removed entirely for a different design so I'm keeping this for now

https://www.lightningdesignsystem.com/components/toast/

File: RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingSlotsContainer/mobileAppointmentBookingSlotsContainer.js
Comment (ibenitamar) [reply RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingSlotsContainer/mobileAppointmentBookingSlotsContainer.js:123]: Those are not the same exactly maxvaliddate and the others are props passed from parent component and onWeekUpdated is a public function called using this.template.querySelector.

Comment (khawkins) [reply RescheduleAppointments/force-app/main/default/lwc/mobileAppointmentBookingSlotsContainer/mobileAppointmentBookingSlotsContainer.js:123]: - All of the components—parents and children—are controlled in this body of source, so in theory the relationships can be changed to best practices.
- To that end, these properties don't fit the convention of property passing between LWC templates and the properties defined in their JS files. Public / `@api` property names between template and code should follow the convention of the property name in code being lower-camel-cased, and its reference in the template should use dash separators ([source](https://developer.salesforce.com/docs/component-library/documentation/en/lwc/lwc.js_props_names)).\
\
For example, `mobileAppointmentBookingLanding.html` should be referencing properties on `mobileAppointmentBookingSchedulingContainer` as follows:

        <c-mobile-appointment-booking-scheduling-container max-valid-date={maxValidCalendarDate} />

    Then you should define the naming of `maxvaliddate` as `maxValidDate` in `mobileAppointmentBookingSchedulingContainer`.

**Pull request merged**
